minai_plugin/customintegrations.php

```diff
@@ -153,6 +159,37 @@ function ProcessIntegrations() {
             )
         );
     }
+    if (isset($GLOBALS["gameRequest"]) && str_starts_with(strtolower($GLOBALS["gameRequest"][0]), "sextalk")) {
+        $type = $GLOBALS["gameRequest"][0];
+        $scene = getScene($GLOBALS["HERIKA_NAME"]);
+        $sceneDesc = $scene["description"];
+        if(!$sceneDesc) {
+            if($scene["fallback"]) {
+                $sceneDesc = $scene["fallback"];
+            } else {
+                $sceneDesc = "{$scene["actors"]} are having sex.";
+            }
+            
+        }
+
+        $prompt = "";
+
+        switch($type) {
+            case "sextalk_scenechange": {
+                $prompt = "The Narrator: ";
+                
+                if(!$scene["prev_scene_id"]) {
+                    $prompt .= "{$scene["actors"]} started sex scene.";
+                } else {
+                    $prompt .= "{$scene["actors"]} changed position.";
+                }
+                $prompt .= " $sceneDesc";
+                break;
+            }
+        }
+
+        $GLOBALS["gameRequest"][3] = $prompt;
```

Comment:
Will we send an empty event in context if the  type is not sextalk_scenechange?

Reply:
yeah, just tackled it in my other branch, when figured that it sends empty string ))
I'll probably merge it here, it has mcm options and other smaller changes

Reply:
merged other commit here which fixes handling cases